tests/beacon/test_helpers.py

```diff
@@ -707,6 +722,224 @@ def test_get_domain(pre_fork_version,
+    )
+
+    assert len(indices) == len(pubkeys)
+
+    for pubkey in pubkeys:
+        assert pubkey in all_pubkeys
+
+
+def _list_and_index(data, max_size=None, elements=st.integers()):
+    '''
+    Hypothesis helper function cribbed from their docs on @composite
+    '''
+    xs = data.draw(st.lists(elements, max_size=max_size))
+    i = data.draw(st.integers(min_value=0, max_value=max(len(xs) - 1, 0)))
+    return (xs, i)
+
+
+@given(st.data())
+def test_generate_aggregate_pubkeys(genesis_validators, sample_slashable_vote_data_params, data):
+    max_value_for_list = len(genesis_validators) - 1
+    (indices, some_index) = _list_and_index(data,
+                                            elements=st.integers(min_value=0,
+                                                                 max_value=max_value_for_list))
+    proof_of_custody_0_indices = indices[:some_index]
+    proof_of_custody_1_indices = indices[some_index:]
+
+    key = "aggregate_signature_poc_0_indices"
+    sample_slashable_vote_data_params[key] = proof_of_custody_0_indices
+    key = "aggregate_signature_poc_1_indices"
+    sample_slashable_vote_data_params[key] = proof_of_custody_1_indices
+
+    votes = SlashableVoteData(**sample_slashable_vote_data_params)
+
+    keys = generate_aggregate_pubkeys(genesis_validators, votes)
+    assert len(keys) == 2
+
+    (poc_0_key, poc_1_key) = keys
+
+    poc_0_keys = get_pubkey_for_indices(genesis_validators, proof_of_custody_0_indices)
+    poc_1_keys = get_pubkey_for_indices(genesis_validators, proof_of_custody_1_indices)
+
+    assert bls.aggregate_pubkeys(poc_0_keys) == poc_0_key
+    assert bls.aggregate_pubkeys(poc_1_keys) == poc_1_key
+
+
+@given(st.data())
+def test_verify_vote_count(max_casper_votes, sample_slashable_vote_data_params, data):
+    (indices, some_index) = _list_and_index(data, max_size=max_casper_votes)
+    proof_of_custody_0_indices = indices[:some_index]
+    proof_of_custody_1_indices = indices[some_index:]
+
+    key = "aggregate_signature_poc_0_indices"
+    sample_slashable_vote_data_params[key] = proof_of_custody_0_indices
+    key = "aggregate_signature_poc_1_indices"
+    sample_slashable_vote_data_params[key] = proof_of_custody_1_indices
+
+    votes = SlashableVoteData(**sample_slashable_vote_data_params)
+
+    assert verify_vote_count(votes, max_casper_votes)
+
+
+def _select_indices(max, count):
+    '''
+    Randomly select some validator indices
+    '''
+    indices = []
+    for i in range(count):
+        next_index = random.randint(0, max - 1)
+        while next_index in indices:
+            next_index = random.randint(0, max - 1)
+        indices.append(next_index)
+    return indices
+
+
+def _get_indices_and_signatures(num_validators,
+                                num_indices,
+                                validators,
+                                message,
+                                privkeys):
+    indices = _select_indices(num_validators, num_indices)
```

Comment:
could be replaced by:
```python
assert num_validators >= num_indices
indices = random.sample(range(num_validators), num_indices)
```

Reply:
ended up going this route -- using hypothesis to sample param space *and* process signatures takes too long